### ISSUE TITLE
Spikekill Backtrace Argument #1 ($string) must be passed by reference, value given

### ISSUE THREAD

**Issue by jdcoats** [Cacti/cacti#5195]
```
2023/01/26 08:06:52 - CMDPHP PHP ERROR WARNING Backtrace: (/cli/removespikes.php[197]:spikekill->remove_spikes(), /lib/spikekill.php[581]:spikekill->calculateVarianceAverages(), /lib/spikekill.php[743]:array_filter(), CactiErrorHandler())
2023/01/26 08:06:52 - ERROR PHP WARNING: spikekill::removeNanFromSamples(): Argument #1 ($string) must be passed by reference, value given in file: /var/www/localhost/htdocs/cacti/lib/spikekill.php on line: 743
2023/01/26 08:06:52 - CMDPHP PHP ERROR WARNING Backtrace: (/cli/removespikes.php[197]:spikekill->remove_spikes(), /lib/spikekill.php[581]:spikekill->calculateVarianceAverages(), /lib/spikekill.php[743]:array_filter(), CactiErrorHandler())
2023/01/26 08:06:52 - ERROR PHP WARNING: spikekill::removeNanFromSamples(): Argument #1 ($string) must be passed by reference, value given in file: /var/www/localhost/htdocs/cacti/lib/spikekill.php on line: 743
2023/01/26 08:06:52 - CMDPHP PHP ERROR WARNING Backtrace: (/cli/removespikes.php[197]:spikekill->remove_spikes(), /lib/spikekill.php[581]:spikekill->calculateVarianceAverages(), /lib/spikekill.php[743]:array_filter(), CactiErrorHandler())
2023/01/26 08:06:52 - ERROR PHP WARNING: spikekill::removeNanFromSamples(): Argument #1 ($string) must be passed by reference, value given in file: /var/www/localhost/htdocs/cacti/lib/spikekill.php on line: 743
2023/01/26 08:06:52 - CMDPHP PHP ERROR WARNING Backtrace: (/cli/removespikes.php[197]:spikekill->remove_spikes(), /lib/spikekill.php[581]:spikekill->calculateVarianceAverages(), /lib/spikekill.php[743]:array_filter(), CactiErrorHandler())
2023/01/26 08:06:52 - ERROR PHP WARNING: spikekill::removeNanFromSamples(): Argument #1 ($string) must be passed by reference, value given in file: /var/www/localhost/htdocs/cacti/lib/spikekill.php on line: 743
2023/01/26 08:06:52 - CMDPHP PHP ERROR WARNING Backtrace: (/cli/removespikes.php[197]:spikekill->remove_spikes(), /lib/spikekill.php[581]:spikekill->calculateVarianceAverages(), /lib/spikekill.php[743]:array_filter(), CactiErrorHandler())
2023/01/26 08:06:52 - ERROR PHP WARNING: spikekill::removeNanFromSamples(): Argument #1 ($string) must be passed by reference, value given in file: /var/www/localhost/htdocs/cacti/lib/spikekill.php on line: 743
```

**Comment by jdcoats** [comment]
Also I have a weird issue where i have scheduled spikekill to run every other day at 10pm yet it runs every other day at 8:05am, I've tried changing the schedule and it always runs at 8am. I think this only started after updating to 1.2.24

**Comment by TheWitness** [comment]
Nothing changed there, but the `array_filter()` use is a bit hard to interpret, so I'll unwind that and use a more well understood looping mechanism for the next guy.

The schedule piece has not change in like forever.  You need to log a separate issue for that.

**Comment by TheWitness** [comment]
Okay, this has been resolved.  Just pull a fresh copy of spikekill.php.  I'll be doing some additional fixes to this in the 1.3 branch, but I'll try to make it possible to backport.
